lib/generators/suspenders/factories_generator.rb

```diff
@@ -0,0 +1,82 @@
+module Suspenders
+  module Generators
+    class FactoriesGenerator < Rails::Generators::Base
+      source_root File.expand_path("../../templates/factories", __FILE__)
+      desc <<~TEXT
+        Build test data with clarity and ease.
+
+        This uses FactoryBot to help you define dummy and test data for your test
+        suite. The `create`, `build`, and `build_stubbed` class methods are directly
+        available to all tests.
+
+        We recommend putting FactoryBot definitions in one `spec/factories.rb` (or
+        `test/factories`) file, at least until it grows unwieldy. This helps reduce
+        confusion around circular dependencies and makes it easy to jump between
+        definitions.
+
+        Supports the default test suite and RSpec.
+      TEXT
+
+      def add_factory_bot
+        gem_group :development, :test do
+          gem "factory_bot_rails"
+        end
+
+        Bundler.with_unbundled_env { run "bundle install" }
+      end
+
+      def set_up_factory_bot
+        if default_test_helper_present?
+          insert_into_file Rails.root.join("test/test_helper.rb"), after: "class TestCase" do
+            "\n    include FactoryBot::Syntax::Methods"
+          end
+        elsif rspec_test_helper_present?
+          copy_file "factory_bot_rspec.rb", "spec/support/factory_bot.rb"
+          insert_into_file Rails.root.join("spec/rails_helper.rb") do
+            %(Dir[Rails.root.join("spec/support/**/*.rb")].sort.each { |file| require file })
+          end
+        end
+      end
+
+      def generate_empty_factories_file
+        if default_test_suite?
+          copy_file "factories.rb", "test/factories.rb"
+        elsif rspec_test_suite?
+          copy_file "factories.rb", "spec/factories.rb"
+        end
+      end
+
+      def remove_fixture_definitions
+        if default_test_helper_present?
+          comment_lines "test/test_helper.rb", /fixtures :all/
+        end
+      end
+
+      def create_linting_test
+        if default_test_suite?
+          copy_file "factories_test.rb", "test/factory_bots/factories_test.rb"
+        elsif rspec_test_suite?
+          copy_file "factories_spec.rb", "spec/factory_bots/factories_spec.rb"
+        end
+      end
+
+      private
+
+      def default_test_suite?
+        File.exist? Rails.root.join("test")
+      end
+
+      def rspec_test_suite?
+        File.exist? Rails.root.join("spec")
```

Comment:
In the `suspenders:ci` generator, we [check against](https://github.com/thoughtbot/suspenders/blob/55cb5c246fe53aea4cf53436c7c208b40ad13e85/lib/suspenders/generators/ci_generator.rb#L29) `spec/spec_helper.rb` instead. I don't know whether one is correct, but I would like to keep this consistent.

Reply:
Good call out. Let's go with `spec/spec_helper.rb`. In an effort to keep things consistent, I extracted that into a helper.

6f436bf